src/Sylius/Bundle/AdminBundle/Controller/RemoveCatalogPromotionAction.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\AdminBundle\Controller;
+
+use Sylius\Bundle\CoreBundle\CatalogPromotion\Processor\CatalogPromotionRemovalProcessorInterface;
+use Sylius\Component\Promotion\Exception\CatalogPromotionNotFoundException;
+use Sylius\Component\Promotion\Exception\InvalidCatalogPromotionStateException;
+use Symfony\Component\HttpFoundation\Exception\BadRequestException;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Session\Session;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+final class RemoveCatalogPromotionAction
+{
+    public function __construct(private CatalogPromotionRemovalProcessorInterface $catalogPromotionRemovalProcessor)
+    {
+    }
+
+    public function __invoke(Request $request): Response
+    {
+        try {
+            $this->catalogPromotionRemovalProcessor->removeCatalogPromotion($request->attributes->get('code'));
+
+            /** @var Session $session */
```

Comment:
The reason I marked it as `Session` instead of `SessionInterface` and spit it into two statements is the lack of `getFlashBag()` method in the `SessionInterface`. Psalm was angry about [it](https://github.com/Sylius/Sylius/actions/runs/3060895779/jobs/4939977662#step:20:43)